lib/register_client_manager.rb

```diff
@@ -33,12 +31,17 @@ def get_register_from_environment(register, environment_url, data_store = nil)
   
       def defaults
         {
+            api_key: nil,
             page_size: 100
         }
       end
 
       def create_register_client(register_url, data_store, page_size)
-        RegistersClient::RegisterClient.new(register_url, data_store, page_size)
+        register_options = {
+            api_key: @config_options.has_key?(:api_key) ? @config_options[:api_key] : nil
```

Comment:
If I'm not missing anything, this can be simplified as

```ruby
register_options = {
  api_key: @config_options[:api_key]
}
```

Reply:
Yep, I think you're right. I spotted this after the PR was made too.